src/hooks/users/useWorldUsers.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { createContext, useContext, useMemo } from "react";
+import { isEqual } from "lodash";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { worldUsersWithoutLocationSelector } from "utils/selectors";
+import { isDefined } from "utils/types";
+
+import { isLoaded, useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+import { useSovereignVenueId } from "hooks/useSovereignVenueId";
+
+const noUsers: WithId<User>[] = [];
+
+export interface WorldUsersContextState {
+  isSovereignVenueIdLoading: boolean;
+  sovereignVenueId?: string;
+  sovereignVenueIdError?: string;
+
+  shouldConnect: boolean;
+}
+
+const WorldUsersContext = createContext<WorldUsersContextState | undefined>(
+  undefined
+);
+
+export interface WorldUsersProviderProps {
+  venueId?: string;
+}
+
+export const WorldUsersProvider: React.FC<WorldUsersProviderProps> = ({
+  venueId,
+  children,
+}) => {
+  const {
+    sovereignVenueId,
+    isSovereignVenueIdLoading,
+    errorMsg: sovereignVenueIdError,
+  } = useSovereignVenueId({
+    venueId,
+  });
+
+  const shouldConnect =
```

Comment:
Not sure if it was obvious for me what this var is about until I got into the logic. Probably should be renamed

Reply:
Given the logic is like literally the next bit of code.. is that such a bad thing?

It seems super straightforward and obvious to me. It controls whether `useFirestoreConnect` `shouldConnect`?

Do you have suggestions for better names?

Reply:
Not a big deal, just sharing my experience. totally NB. I don't have the suggestions though

Reply:
Fair fair. Will call this 'wont fix' then, unless someone brings up a better name/etc.